cupy/array_api/_array_object.py

```diff
@@ -442,7 +441,7 @@ def __dlpack_device__(self: Array, /) -> Tuple[IntEnum, int]:
         """
         return self._array.__dlpack_device__()
 
-    def __eq__(self: Array, other: Union[int, float, bool, Array], /) -> Array:
+    def __eq__(self: Array, other: Union[int, float, bool, Array], /) -> Array:  # type: ignore
```

Comment:
Yeah, this is an unfortunate side-effect of being incompatible with the (somewhat overly strict) [`object.__eq__`/`__ne__`](https://github.com/python/typeshed/blob/9687d53b65d055a990f0187b7baa1137cb7c717b/stdlib/builtins.pyi#L102-L103) annotations. Nothing to do about this though except adding a `# type: ignore` comment or complain a bit more upstream...